x-pack/plugins/cloud_security_posture/public/pages/rules/rules_table.tsx

```diff
@@ -0,0 +1,137 @@
+} from '@elastic/eui';
+// import moment from 'moment';
+import type { RulesState } from './rules_container';
+import * as TEST_SUBJECTS from './test_subjects';
+import * as TEXT from './translations';
+import type { RuleSavedObject } from './use_csp_rules';
+
+type RulesTableProps = Pick<
+  RulesState,
+  'loading' | 'error' | 'rules_page' | 'total' | 'perPage' | 'page'
+> & {
+  toggleRule(rule: RuleSavedObject): void;
+  setSelectedRules(rules: RuleSavedObject[]): void;
+  setPagination(pagination: Pick<RulesState, 'perPage' | 'page'>): void;
+  // ForwardRef makes this ref not available in parent callbacks
+  tableRef: React.RefObject<EuiBasicTable<any>>;
+};
+
+export const RulesTable = ({
+  toggleRule,
+  setSelectedRules,
+  setPagination,
+  perPage: pageSize,
+  rules_page: items,
+  page,
+  tableRef,
+  total,
+  loading,
+  error,
+}: RulesTableProps) => {
+  const columns = useMemo(() => getColumns({ toggleRule }), [toggleRule]);
+
+  const euiPagination: EuiBasicTableProps<RuleSavedObject>['pagination'] = {
+    pageIndex: page,
+    pageSize,
+    totalItemCount: total,
+    pageSizeOptions: [1, 5, 10, 25],
+    hidePerPageOptions: false,
+  };
+
+  const selection: EuiBasicTableProps<RuleSavedObject>['selection'] = {
+    selectable: () => true,
+    onSelectionChange: setSelectedRules,
+  };
+
+  const onTableChange = ({ page: pagination }: Criteria<RuleSavedObject>) => {
+    if (!pagination) return;
+    setPagination({ page: pagination.index, perPage: pagination.size });
+  };
+
+  return (
+    <EuiBasicTable
+      ref={tableRef}
+      data-test-subj={TEST_SUBJECTS.CSP_RULES_TABLE}
+      loading={loading}
+      error={error}
+      items={items}
+      columns={columns}
+      pagination={euiPagination}
+      onChange={onTableChange}
+      isSelectable={true}
+      selection={selection}
+      itemId={(v) => v.id}
+    />
+  );
+};
+
+const ruleNameRenderer = (name: string) => (
+  <EuiLink className="eui-textTruncate" title={name}>
+    {name}
+  </EuiLink>
+);
+
+// const timestampRenderer = (timestamp: string) =>
+//   moment.duration(moment().diff(timestamp)).humanize();
+
+interface GetColumnProps {
+  toggleRule: (rule: RuleSavedObject) => void;
+}
+
+const createRuleEnabledSwitchRenderer =
+  ({ toggleRule }: GetColumnProps) =>
+  (value: any, rule: RuleSavedObject) =>
+    (
+      <EuiSwitch
+        showLabel={false}
+        label={value ? TEXT.DISABLE : TEXT.ENABLE}
+        checked={value}
+        onChange={() => toggleRule(rule)}
+        data-test-subj={TEST_SUBJECTS.getCspRulesTableItemSwitchTestId(rule.attributes.id)}
+      />
+    );
+
+const getColumns = ({
+  toggleRule,
+}: GetColumnProps): Array<EuiTableFieldDataColumnType<RuleSavedObject>> => [
+  {
+    field: 'attributes.name',
+    name: TEXT.RULE_NAME,
+    width: '60%',
+    truncateText: true,
+    render: ruleNameRenderer,
+  },
+  {
+    field: 'section', // TODO: what field is this?
+    name: TEXT.SECTION,
+    width: '15%',
```

Comment:
this doesn't exist on the pre-installed rules. should be added when we map CIS benchmark to our rules.